pkg/proxy/proxy.go

```diff
@@ -411,12 +411,12 @@ func (p *Proxy) ReinstallRoutingRules() error {
 		if err := removeFromEgressProxyRoutesIPv4(); err != nil {
 			return err
 		}
-		if !option.Config.EnableIPSec || option.Config.TunnelingEnabled() {
-			if err := removeFromIngressProxyRoutesIPv4(); err != nil {
+		if (option.Config.EnableEnvoyConfig || option.Config.EnableIPSec) && !option.Config.TunnelingEnabled() {
```

Comment:
We're switching the cases here, is the intention to do the opposite expression?

i.e. if we do this only if `!option.Config.TunnelingEnabled()` where previously the "else" case could happen with  TunnelingEnabled being either true/false but now we install only if tunneling is disabled.

Reply:
it's intentional as per the below convo https://github.com/cilium/cilium/pull/31280#discussion_r1528020569. 

(PS: I keep the above  PR to debug another issue)